crates/sui-benchmark/src/lib.rs

```diff
@@ -231,6 +290,147 @@ impl ValidatorProxy for LocalValidatorAggregatorProxy {
+                }
+
+                let signatures: BTreeMap<_, _> = votes
+                    .into_iter()
+                    .map(|a| (a.authority, a.signature))
+                    .collect();
+                let mut signers_map = RoaringBitmap::new();
+                for pk in signatures.keys() {
+                    signers_map.insert(
+                        self.committee
+                            .authority_index(pk)
+                            .ok_or(SuiError::UnknownSigner {
+                                signer: Some(pk.concise().to_string()),
+                                index: None,
+                                committee: self.committee.clone(),
+                            })
+                            .expect("Received signature from unknown validator")
+                            as u32,
+                    );
+                }
+                let sigs: Vec<AuthoritySignature> = signatures.into_values().collect();
+
+                let quorum_signature = AuthorityQuorumSignInfo {
+                    epoch,
+                    // Note: This function simply aggregates signatures (it does not check that they
+                    // are individually valid).
+                    signature: AggregateAuthoritySignature::aggregate(&sigs)
+                        .map_err(|e| SuiError::InvalidSignature {
+                            error: e.to_string(),
+                        })
+                        .expect("Validator returned invalid signature"),
+                    signers_map,
+                };
+
+                Envelope::new_from_data_and_sig(message_data.unwrap(), quorum_signature)
+            }
+        };
+
+        // Send the certificate to all validators.
+        let mut futures = FuturesUnordered::new();
+        total_stake = 0;
+        let mut transaction_effects = None;
+        for client in self.clients.values() {
+            let fut = client.handle_certificate(certified_transaction.clone());
+            futures.push(fut);
+        }
+
+        // Wait for the replies from a quorum of validators.
+        while let Some(response) = futures.next().await {
+            match response {
+                // If all goes well, the validators reply with signed effects.
+                Ok(HandleCertificateResponse { signed_effects }) => {
+                    let author = signed_effects.auth_sig().authority;
+                    transaction_effects = Some(signed_effects.data().clone());
+                    total_stake += self.committee.weight(&author);
+                }
+
+                // This typically happens when the validators are overloaded and the certificate is
+                // immediately rejected.
+                Err(e) => tracing::warn!("Failed to submit certificate: {e}"),
+            }
+
+            // TODO: We can't stop sending transactions after receiving a quorum of replies otherwise
```

Comment:
What exactly is the TODO here? IIUC you are commented the code out to stop sending TX after receiving a quorum of replies below. So this TODO is done?

Also is the issue that if we do stop here that other validators that we didnt send the TX to will have to wait for local execution before processing the next TX?

Reply:
Good point. The comment is unclear. It would be better to compute latency once we receive a quorum of replies rather than all replies (also because this will prevent us from running benchmarks under faults). I will clarify the comment and address that in a separate PR